src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/lifetime-connections.md

```diff
@@ -0,0 +1,98 @@
+---
+minutes: 20
+---
+
+# Lifetime "Connections" & External Resources
+
+Using `PhantomData` in conjunction with lifetimes lets us say "this value may
+own its data, but it can only live as long as the value that generated it" in
+rust's type system.
+
+```rust,editable,compile_fail
+use std::marker::PhantomData;
+
+pub struct BorrowedButOwned<'a> {
```

Comment:
This example (StringOrigin, BorrowedButOwned) is abstract and a bit confusing. It's not immediately clear what real-world problem this code could solve. The names are generic, and the act of cloning a string to create a "borrowed" value feels counter-intuitive without a concrete motivation (even though that's exactly what would happen in a real-world application!)

I suggest to rewrite this slide to be the motivation and explanation of using PhantomData with generic lifetime parameters (base it upon my suggested `struct Transaction(DatabaseHandle)` example). Leave the `phantomdata.md` slide to be purely about using PhantomData with generic type parameters.